app/src/main/java/pl/futurecollars/invoicing/db/sql/SqlInvoiceRepository.java

```diff
@@ -0,0 +1,288 @@
+                sellerId,
+                invoice.getSeller().getAddress().getCity(),
+                invoice.getSeller().getAddress().getPostalCode(),
+                invoice.getSeller().getAddress().getStreetName(),
+                invoice.getSeller().getAddress().getStreetNumber()
+            );
+
+            jdbcTemplate.update(
+                "insert into address (company_id, city, postal_code, street_name, street_number) values (?, ?, ?, ?, ?);",
+                buyerId,
+                invoice.getBuyer().getAddress().getCity(),
+                invoice.getBuyer().getAddress().getPostalCode(),
+                invoice.getBuyer().getAddress().getStreetName(),
+                invoice.getBuyer().getAddress().getStreetNumber()
+            );
+
+            jdbcTemplate.update(
+                "insert into invoice (id, date_at, invoice_number, seller, buyer) values (?, ?, ?, ?, ?);",
+                invoiceId,
+                java.sql.Date.valueOf(invoice.getDateAt()),
+                invoice.getNumber(),
+                sellerId,
+                buyerId
+            );
+
+            invoice.getEntries().forEach(entry -> {
+                UUID invoiceEntryId = UUID.randomUUID();
+                UUID carId = UUID.randomUUID();
+                jdbcTemplate.update("insert into car (id, registration_number, personal_usage) values (?, ?, ?);",
+                    carId,
+                    entry.getCarRelatedExpense().getRegistrationNumber(),
+                    entry.getCarRelatedExpense().isPersonalUsage()
+                );
+                jdbcTemplate.update("insert into invoice_entry (id, description, price, vat_value, vat_rate, car_related_expense, invoice) "
+                        + "values (?, ?, ?, ?, ?, ?, ?);",
+                    invoiceEntryId,
+                    entry.getDescription(),
+                    entry.getPrice(),
+                    entry.getVatValue(),
+                    vatToId.get(entry.getVatRate()),
+                    carId,
+                    invoiceId
+                );
+            });
+
+            return invoice;
+        } else {
+            throw new DbException("ID already exist.");
+        }
+    }
+
+    @Override
+    public Optional<Invoice> getById(UUID id) {
+        List<Invoice> invoices = jdbcTemplate.query(SELECT_FROM_INVOICE_QUERY + " where i.id = '" + id + "'", invoiceRowMapper());
+        return invoices.isEmpty() ? Optional.empty() : Optional.of(invoices.get(0));
+    }
+
+    @Override
+    public List<Invoice> getAll() {
+        return jdbcTemplate.query(SELECT_FROM_INVOICE_QUERY, invoiceRowMapper());
+    }
+
+    private RowMapper<Invoice> invoiceRowMapper() {
+        return (rs, rowNr) -> {
+            UUID invoiceId = UUID.fromString(rs.getString("id"));
+            List<InvoiceEntry> invoiceEntries = jdbcTemplate.query(
+                "select * from invoice_entry "
```

Comment:
It is better to write query for specific columns instead of *

Reply:
Done.